### PR TITLE
Upgrade the generator script to use Clang.jl's new generator

### PR DESCRIPTION
**Pull request by Gnimuc**
TODO:
- [x] Update the generator script
- [x] Enable macro support

### CODE REVIEW

**Comment by SimonDanisch** [comment]
Awesome! I guess this still needs fixing the tests?

**Comment by Gnimuc** [comment]
I enabled the test for AbstractFreeType, so the test would not pass. Is there another way to embed the testing of a downstream package into the parent package?

**Comment by SimonDanisch** [comment]
Oh I see.. I think that's mostly not a good idea :D I've had the urge to do this multiple times and got rid of it later all the time...

If there are tests we really want to run in here to make sure FreeTypeAbstraction work, we should move those tests to FreeType.jl ...

**Comment by Gnimuc** [comment]
I feel the same. That line is left commented for now. But I guess someone will mistakenly enable it again in the future. 😄

File: src/LibFreeType.jl [hunk 1]
```diff
@@ -15,14 +18,1505 @@ const LONG_MAX = typemax(Clong)
...
+
+struct FT_ListNodeRec_
+    prev::FT_ListNode
+    next::FT_ListNode
+    data::Ptr{Cvoid}
+end
+
+const FT_ListNodeRec = FT_ListNodeRec_
+
+const FT_ListRec = FT_ListRec_
```
Comment (Gnimuc) [inline src/LibFreeType.jl]: Another thing I'd like to point out is that forward-decls are handled in a new way with Clang.jl's new generator.

**Comment by SimonDanisch** [comment]
Is this ready? Should we tag this as breaking, so we can upgrade in peace and make sure all dependencies work?

**Comment by Gnimuc** [comment]
> Should we tag this as breaking, so we can upgrade in peace and make sure all dependencies work?

Do you mean to bump the major version? If we're going to bump the major version, I think it's a perfect time to remove the `compat.jl` and export everything just like `ModernGL.jl`. In this way, `LibFreeType.jl` can also be renamed to `FreeType.jl`.

**Comment by SimonDanisch** [comment]
Yeah, that sounds like a plan!

**Comment by Gnimuc** [comment]
Merged and tagged!

**Pull request merged**